pkg/model_registry/service/registered_models.go

```diff
@@ -0,0 +1,190 @@
+	if err != nil {
+		return nil, err
+	}
+
+	return &protos.RenameRegisteredModel_Response{
+		RegisteredModel: registeredModel.ToProto(),
+	}, nil
+}
+
+func (m *ModelRegistryService) DeleteRegisteredModel(
+	ctx context.Context, input *protos.DeleteRegisteredModel,
+) (*protos.DeleteRegisteredModel_Response, *contract.Error) {
+	if err := m.store.DeleteRegisteredModel(ctx, input.GetName()); err != nil {
+		return nil, err
+	}
+
+	return &protos.DeleteRegisteredModel_Response{}, nil
+}
+
+func (m *ModelRegistryService) GetRegisteredModel(
+	ctx context.Context, input *protos.GetRegisteredModel,
+) (*protos.GetRegisteredModel_Response, *contract.Error) {
+	registeredModel, err := m.store.GetRegisteredModel(ctx, input.GetName())
+	if err != nil {
+		return nil, err
+	}
+
+	return &protos.GetRegisteredModel_Response{
+		RegisteredModel: registeredModel.ToProto(),
+	}, nil
+}
+
+func (m *ModelRegistryService) SetRegisteredModelTag(
+	ctx context.Context, input *protos.SetRegisteredModelTag,
+) (*protos.SetRegisteredModelTag_Response, *contract.Error) {
+	name := input.GetName()
+	if name == "" {
+		return nil, contract.NewError(
+			protos.ErrorCode_INVALID_PARAMETER_VALUE,
+			"Registered model name cannot be empty",
+		)
+	}
+
+	if err := m.store.SetRegisteredModelTag(ctx, name, input.GetKey(), input.GetValue()); err != nil {
+		return nil, err
+	}
+
+	return &protos.SetRegisteredModelTag_Response{}, nil
+}
+
+func (m *ModelRegistryService) CreateRegisteredModel(
+	ctx context.Context, input *protos.CreateRegisteredModel,
+) (*protos.CreateRegisteredModel_Response, *contract.Error) {
+	name := input.GetName()
+	if name == "" {
+		return nil, contract.NewError(
+			protos.ErrorCode_INVALID_PARAMETER_VALUE,
+			"Registered model name cannot be empty.",
+		)
+	}
+
+	tags := make([]*entities.RegisteredModelTag, 0, len(input.GetTags()))
+	for _, tag := range input.GetTags() {
+		tags = append(tags, entities.NewRegisteredModelTagFromProto(tag))
+	}
+
+	registeredModel, err := m.store.CreateRegisteredModel(ctx, input.GetName(), input.GetDescription(), tags)
+	if err != nil {
+		return nil, err
+	}
+
+	return &protos.CreateRegisteredModel_Response{
+		RegisteredModel: registeredModel.ToProto(),
+	}, nil
+}
+
+func (m *ModelRegistryService) DeleteRegisteredModelTag(
+	ctx context.Context, input *protos.DeleteRegisteredModelTag,
+) (*protos.DeleteRegisteredModelTag_Response, *contract.Error) {
+	name := input.GetName()
+	if name == "" {
+		return nil, contract.NewError(
+			protos.ErrorCode_INVALID_PARAMETER_VALUE,
+			"Registered model name cannot be empty",
+		)
+	}
+
+	if err := m.store.DeleteRegisteredModelTag(ctx, name, input.GetKey()); err != nil {
+		return nil, err
+	}
+
+	return &protos.DeleteRegisteredModelTag_Response{}, nil
+}
+
+func (m *ModelRegistryService) SetRegisteredModelAlias(
+	ctx context.Context, input *protos.SetRegisteredModelAlias,
+) (*protos.SetRegisteredModelAlias_Response, *contract.Error) {
+	name := input.GetName()
```

Comment:
I don't quite get these validation checks for empty values?
I would expect that the annotations cover this, no?

Reply:
Thanks!